Added support for dynamic building builder

ePasha opened the pull request.

xperiandri commented on src/Elmish.Uno/Binding.fs:
```diff
@@ -1749,6 +1749,13 @@ type Binding private () =
     Binding.SubModel.required bindings
     >> Binding.mapModel (fun m -> (m, getSubModel m))
 
+  static member subModelWithModel
+      (getSubModel: 'model -> 'subModel,
+       bindings: Binding<'subModel, 'msg> list)
+      : string -> Binding<'model, 'msg> =
+    Binding.SubModel.required bindings
+    >> Binding.mapModel (fun m -> (getSubModel m))
```
Всё наоборот. Нужно те существующие методы где `'model * 'submodel` переименовать на `subModelWithModel`
А под это сделать новые парные методы под каждым из тех, в которых чисто `'submodel`

xperiandri commented on src/Elmish.Uno/Binding.fs:
```diff
@@ -1742,21 +1742,42 @@ type Binding private () =
     Binding.SubModel.required bindings
     >> Binding.mapModel (fun m -> (m, getSubModel m))
 
+  static member subModelWithModel
```
А здесь XML комментарий?

xperiandri commented on src/Elmish.Uno/Binding.fs:
```diff
@@ -1659,7 +1659,7 @@ type Binding private () =
   /// </param>
   /// <param name="bindings">The bindings for the sub-model.</param>
-  static member subModel
+  static member subModelWithModel
```
На каждое переименование сразу под ним должен быть новый соответствующий метод со старым названием и парамерами `'subModel` вместо `'model * 'subModel`

xperiandri reviewed (changes requested):
Have you verified that `Samples.sln` build?

xperiandri committed:
Aligned `Binding.subModel` and `Binding.subModelWithModel` to `BindingT.subModel` and `BindingT.subModelWithModel`

Renamed `Binding.subModel` to `Binding.subModelWithModel` and added `Binding.subModel` overloads

ePasha committed:
Added `.idea/` to `.gitignore`

ePasha committed:
Added support for dynamic building builder
